app/src/main/java/com/rob729/quiethours/Adapter/ProfileListAdapter.kt

```diff
@@ -65,12 +69,60 @@ class ProfileListAdapter(
             binding.TxtImg.setText(item.name[0].toString())
             binding.TxtImg.avatarBackgroundColor = bgColors[item.colorIndex]
             binding.timeStamp.text = item.timeInstance
+            binding.pauseSwitch.isChecked = item.pauseSwitch
+            val current: Calendar = Calendar.getInstance()
+            val currentHour = current.get(Calendar.HOUR_OF_DAY)
+            val currentMinute = current.get(Calendar.MINUTE)
 
             binding.profileCard.setOnClickListener {
                 val args = Bundle()
                 args.putParcelable("Profile", item)
                 val dialog = DetailsFragment.newInstance(args)
-                dialog.show((parentView.context as FragmentActivity).supportFragmentManager, "DialogFragment")
+                dialog.show(
+                    (parentView.context as FragmentActivity).supportFragmentManager,
+                    "DialogFragment"
+                )
+            }
+            if (item.ehr == currentHour && item.emin == currentMinute
+            ) {
+                StoreSession.writeInt(AppConstants.BEGIN_STATUS, 0)
+                StoreSession.writeLong(AppConstants.ACTIVE_PROFILE_ID, 0)
+            }
+            if (item.shr == currentHour && item.smin == currentMinute
+            ) {
+                StoreSession.writeLong(AppConstants.ACTIVE_PROFILE_ID, item.profileId)
+            }
+            binding.pauseSwitch.setOnClickListener {
+                item.pauseSwitch = false
+                WorkManagerHelper.cancelWork(item.profileId.toString())
+                if (!binding.pauseSwitch.isChecked) {
+                    if (StoreSession.readLong(AppConstants.ACTIVE_PROFILE_ID) == item.profileId)
+                        audioManager.ringerMode = StoreSession.readInt(AppConstants.RINGTONE_MODE)
+                } else {
+                    if ((StoreSession.readLong(AppConstants.ACTIVE_PROFILE_ID) == item.profileId) && ((item.shr == item.ehr && (currentHour == item.ehr && currentMinute > item.emin) || (currentHour > item.ehr)) || (item.shr < item.ehr && (currentHour >= item.ehr)) || (item.shr > item.ehr && (currentHour <= item.ehr)))) {
```

Comment:
Have you autoformatted the code using Ctrl+Alt+L in this file as the above expression is quite long and still in one line?

Reply:
Autoformatted.....because first I wrote the whole code and then used Ctrl+Alt+L in ProfileListAdapter.kt

I rechecked it now by again using Ctrl+Alt+L, an it remains the same.....in one line

Reply:
Okay.
Please break this line into 2/3 lines for better readability.

Reply:
@rob729 Done, I first broke the code in 2-3 lines, then did Ctrl+Alt+L to reformat it